pip/index.py

```diff
@@ -1184,3 +1190,13 @@ def is_wheel(self):
 # An object to represent the "link" for the installed version of a requirement.
 # Using Inf as the url makes it sort higher.
 INSTALLED_VERSION = Link(Inf)
+
+
+Search = namedtuple('Search', 'user canonical formats')
+"""Capture key aspects of a search.
+
+:attribute user: The user supplied package.
```

Comment:
Not sure about the `user` attribute, I had to come here to the `Search` definition to check what the meaning was.
Not a lot of better names though, maybe `supplied_name`, `searched_name` ?